pkg/state/transaction_checker.go

```diff
@@ -333,6 +336,138 @@ func (tc *transactionChecker) checkTransferWithSig(transaction proto.Transaction
 	return smartAssets, nil
 }
 
+func (tc *transactionChecker) checkEthereumTransactionWithProofs(transaction proto.Transaction, info *checkerInfo) ([]crypto.Digest, error) {
+	metamaskActivated, err := tc.stor.features.newestIsActivated(int16(settings.RideV6))
+	if err != nil {
+		return nil, errors.Errorf("failed to check whether feature %d was activated: %v", settings.RideV6, err)
+	}
+	if !metamaskActivated {
+		return nil, errors.Errorf("failed to handle ethereum transaction: feature %d has not been activated yet", settings.RideV6)
+	}
+
+	tx, ok := transaction.(*proto.EthereumTransaction)
+	if !ok {
+		return nil, errors.New("failed to convert interface to TransferWithSig transaction")
+	}
+
+	if tx.EthereumTxType() != proto.EthereumLegacyTxType {
+		return nil, errors.New("the ethereum transaction's type is not legacy tx")
+	}
+	// a cancel transaction
+	// value == 0 && data == 0x
+	if tx.Value().Cmp(big.NewInt(0)) == 0 && len(tx.Data()) == 0 {
+		return nil, errors.New("canceling a transaction is forbidden")
+	}
+
+	// gasPrice == 10
+	if tx.GasPrice().Cmp(big.NewInt(int64(proto.EthereumGasPrice))) != 0 {
+		return nil, errors.Errorf("Gas price should be %d, got %d", int64(proto.EthereumGasPrice), tx.GasPrice().Int64())
```

Comment:
Properly format error message

Reply:
Fixed and removed. You can find this check in 'EthereumTransaction.Validate' method.